compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -2102,6 +2108,47 @@ fn default_to_whole_archive(sess: &Session, crate_type: CrateType, cmd: &dyn Lin
         && !(sess.target.limit_rdylib_exports && cmd.exported_symbol_means_used_symbol())
 }
 
+// Write the debugger visualizer files for each crate to the temp directory and gather the file paths.
+fn collect_debugger_visualizers(
+    tmpdir: &Path,
+    sess: &Session,
+    crate_info: &CrateInfo,
+) -> Vec<PathBuf> {
+    let mut visualizer_paths = Vec::new();
+    let debugger_visualizers = &crate_info.debugger_visualizers;
+    let mut index = 0;
+
+    for (cnum, visualizers) in debugger_visualizers {
```

Comment:
if you make this `for (&cnum, visualizers) in ...` then you won't need the `&` in `&LOCAL_CRATE` below. Alternatively, you can remove the `&`  in `crate_info.crate_name[&cnum].as_str()`